contracts/examples/erc1155/src/lib.rs

```diff
@@ -0,0 +1,387 @@
+		let creator = self.get_token_type_creator(&type_id);
+
+		require!(
+			self.get_caller() == creator,
+			"Only the token creator may mint more tokens"
+		);
+
+		self.increase_balance(&creator, &type_id, &amount);
+
+		if self.get_is_fungible(&type_id) == false {
+			let id_first = self.get_last_valid_token_id_for_type(&type_id);
+			let id_last = id_first.clone() + amount;
+
+			self.set_owner_for_range(&type_id, &id_first, &id_last, &creator);
+
+			self.set_last_valid_token_id_for_type(&type_id, &id_last);
+		}
+
+		// self.transfer_single_event(&caller, &from, &to, &id, &amount);
+
+		Ok(())
+	}
+
+	#[endpoint]
+	fn burn(&self, type_id: BigUint, amount: BigUint) -> SCResult<()> {
+		require!(
+			self.get_is_fungible(&type_id) == true,
+			"Only fungible tokens can be burned"
+		);
+
+		let caller = self.get_caller();
+		let balance = self
+			.get_balance_mapper(&caller)
+			.get(&type_id)
+			.unwrap_or_else(|| BigUint::zero());
+
+		require!(balance >= amount, "Not enough tokens to burn");
+
+		self.decrease_balance(&caller, &type_id, &amount);
+
+		Ok(())
+	}
+
+	// views
+
+	#[view(supportsInterface)]
+	fn supports_interface(&self, interface_id: u32) -> bool {
+		interface_id == INTERFACE_SIGNATURE_ERC165 || interface_id == INTERFACE_SIGNATURE_ERC1155
+	}
+
+	#[view(balanceOf)]
+	fn balance_of(&self, owner: &Address, type_id: &BigUint) -> BigUint {
+		self.get_balance_mapper(&owner)
+			.get(&type_id)
+			.unwrap_or_else(|| BigUint::zero())
+	}
+
+	// returns balance for each (owner, id) pair
+	#[view(balanceOfBatch)]
+	fn balance_of_batch(&self, owners: &[Address], type_ids: &[BigUint]) -> Vec<BigUint> {
+		let mut batch_balance = Vec::new();
+
+		for i in 0..owners.len() {
+			batch_balance.push(self.balance_of(&owners[i], &type_ids[i]));
+		}
+
+		batch_balance
+	}
+
+	#[view(isApprovedForAll)]
+	fn is_approval_for_all(&self, owner: Address, operator: Address) -> bool {
+		self.get_is_approved(&operator, &owner)
+	}
+
+	// private
+
+	// mock
+	fn is_smart_contract_address(&self, _address: &Address) -> bool {
+		false
+	}
+
+	fn is_valid_type_id(&self, type_id: &BigUint) -> bool {
+		type_id > &0 && type_id <= &self.get_last_valid_type_id()
+	}
+
+	fn is_valid_token_id(&self, type_id: &BigUint, token_id: &BigUint) -> bool {
+		self.is_valid_type_id(type_id)
+			&& token_id > &0
+			&& token_id <= &self.get_last_valid_token_id_for_type(type_id)
+	}
+
+	fn increase_balance(&self, owner: &Address, type_id: &BigUint, amount: &BigUint) {
+		let mut balance_mapper = self.get_balance_mapper(owner);
+		let mut balance = balance_mapper
+			.get(type_id)
+			.unwrap_or_else(|| BigUint::zero());
+
+		balance += amount;
+		balance_mapper.insert(type_id.clone(), balance);
+	}
+
+	fn decrease_balance(&self, owner: &Address, type_id: &BigUint, amount: &BigUint) {
+		let mut balance_mapper = self.get_balance_mapper(owner);
+		let mut balance = balance_mapper.get(type_id).unwrap();
```

Comment:
unwrap panics, please avoid

I think we could use some `get_or_default` method in the mapper, it seems to be often necessary

Reply:
I think unwrap makes the wasm code bigger so it should be avoided. if you have get_or_default you can still check if it is the default value or something was written there.